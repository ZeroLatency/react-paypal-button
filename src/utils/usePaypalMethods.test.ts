import { renderHook } from '@testing-library/react-hooks';
import { cleanup } from '@testing-library/react';
import { usePaypalMethods } from './usePaypalMethods';
import { PayPalButtonProps } from '../types';

const baseProps: PayPalButtonProps = {
  amount: 1.00,
  paypalOptions: {
    clientId: '12345',
    intent: 'capture'
  },
  onPaymentStart: jest.fn()
}

describe('usePaypalMethods', () => {
  afterEach(cleanup)
  it('returns an object with methods for paypal lib', () => {
    const { result } = renderHook(() => usePaypalMethods(baseProps))
    expect(result.current).toMatchSnapshot()
  });

  it('returns null for createSubscription if vault prop is false', () => {
    const { result } = renderHook(() => usePaypalMethods(baseProps))
    expect(result.current.createSubscription).toBeNull()
  })

  it('returns null for createOrder if vault prop is true', () => {
    const props = {
      ...baseProps,
      paypalOptions: {
        ...baseProps.paypalOptions,
        vault: true
      }
    }
    const { result } = renderHook(() => usePaypalMethods(props))
    expect(result.current.createOrder).toBeNull()
  })
})
