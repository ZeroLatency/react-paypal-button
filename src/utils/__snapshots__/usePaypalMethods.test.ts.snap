// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePaypalMethods returns an object with methods for paypal lib 1`] = `
Object {
  "createOrder": [Function],
  "createSubscription": [Function],
  "onApprove": [Function],
  "onCancel": [Function],
  "onError": [Function],
  "onShippingChange": [Function],
  "payment": [Function],
}
`;
